subprojects/base-services/src/integTest/groovy/org/gradle/internal/operations/BuildOperationWorkerRegistryIntegrationTest.groovy

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.operations
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class BuildOperationWorkerRegistryIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
We shouldn't have int tests that simply exercise internal services. Instead, the test should exercise some public thing that a user would use - let's say resolving stuff at configuration time (without the workarounds that we added) or using the worker API at configuration time.